src/main/java/blackjack/domain/participant/Player.java

```diff
@@ -6,34 +6,50 @@
 
 public class Player extends Participant {
 
-    public Player(String name, Hand cardHand) {
+    private Money bettingMoney = Money.ZERO;
+
+    public Player(Name name, Hand cardHand) {
+        super(name, cardHand);
+    }
+
+    public Player(Name name, Hand cardHand, Money bettingMoney) {
         super(name, cardHand);
+        this.bettingMoney = bettingMoney;
     }
 
-    public static Player from(String name) {
-        return new Player(name, Hand.createEmptyHand());
+    public static Player from(Name name, Money bettingMoney) {
+        return new Player(name, Hand.createEmptyHand(), bettingMoney);
     }
 
     public void receiveCard(Card card) {
         cardHand.add(card);
     }
 
-    public MatchResult getMatchResult(int dealerCardSum) {
-        if (isBust() || getHandTotal() < dealerCardSum) {
-            return MatchResult.LOSE;
+    public Money getWinningMoney(Dealer dealer) {
+        if (blackjackWin(dealer)) {
+            return bettingMoney.multiplyByRate(MatchResult.BLACKJACK.getProfitRate());
         }
-        if (dealerCardSum == getHandTotal()) {
-            return MatchResult.TIE;
+        if (lose(dealer)) {
+            return bettingMoney.toNegative();
         }
-        if (dealerCardSum < getHandTotal()) {
-            return MatchResult.WIN;
+        if (tie(dealer)) {
+            return Money.ZERO;
         }
+        return bettingMoney.multiplyByRate(MatchResult.STAY.getProfitRate());
+    }
+
+    private boolean blackjackWin(Dealer dealer) {
+        return isBlackjack() && !dealer.isBlackjack();
+    }
 
-        throw new IllegalArgumentException("입력값이 잘못되었습니다. 승무패를 계산할 수 없습니다.");
+    private boolean lose(Dealer dealer) {
+        return isBust()
+                || (dealer.isBlackjack() && !isBlackjack())
+                || (getScore() < dealer.getScore());
     }
```

Comment:
```suggestion
    private boolean lose(Dealer dealer) {
        if (isBust()) {
            return true;
        }
        if (dealer.isBlackjack() && !isBlackjack()) {
            return true;
        }
        return (getScore() < dealer.getScore());
    }
```

코드 라인을 줄이는 것도 좋지만 위처럼 작성하는게 어떤 경우 지는지 알기 쉽지 않을까요?

이번과는 크게 상관없지만 이김에 봐두면 좋을 것 같아 Early Return에 대한 포스팅 공유드립니다. 😉 
[Early Return 하는 코드를 작성하자](https://jheloper.github.io/2019/06/write-early-return-code/)

Reply:
와 읽기 좋네요!! 둘다 셋다 true를 리턴하니 같은 걸 리턴하는 조건들은 모아야한다고 생각했는데 구문이 짧게 끝나는게 훨씬 쉽게 읽히는군요ㅎㅎ 포스팅도 정말 좋은 내용이에요! 감사합니다 🙏